core/src/test/scala/io/chrisdavenport/shellfish/FileOsSpec.scala

```diff
@@ -0,0 +1,91 @@
+package io.chrisdavenport.shellfish
+
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+import org.scalacheck.Gen
+
+import cats.effect.IO
+
+import fs2.io.file.{Path, Files}
+
+import syntax.path.*
+
+object FileOsSpec extends SimpleIOSuite with Checkers {
+
+  val stringGen: Gen[String] = Gen.asciiStr
+  val contents: String =
+    stringGen.sample.getOrElse("Can't gen the contents 🤷🏻‍♀️")
+
+  test("The API should create and delete a file") {
+    val t = Path("core/src/test/resources/temp.tmp")
+    for {
+      _       <- t.createFile
+      _       <- t.write(contents)
+      deleted <- t.deleteIfExists
+    } yield expect(deleted)
+  }
+
+  test("Copying a file should have the same content as the original") {
+
+    val t1 = Path("core/src/test/resources/temp1.tmp")
+    val t2 = Path("core/src/test/resources/temp2.tmp")
+
+    for {
+      _  <- t1.createFile
+      _  <- t1.write(contents)
+      _  <- t1.copy(t2)
+      s1 <- t1.read
+      s2 <- t2.read
+      _  <- t1.deleteIfExists
+      _  <- t2.deleteIfExists
+
+    } yield expect.same(s1, s2)
+  }
+
+  test(
+    "Append to a file using `append` should increate the size of the file in one"
+  ) {
+
+    val temp = Files[IO].tempFile
+    val contentGenerator =
+      Gen.size.flatMap(size => Gen.listOfN(size, stringGen))
```

Comment:
Thanh do you know if the `forall` function evaluates cases in parallel? I was having problems when I used them but it could be due to the fact I didn't used the temp files API to begin with 🤔

Reply:
yes it runs in parallel by default, but We can config it as https://github.com/disneystreaming/weaver-test/blob/5a61ebed194578a8b741ace1c96f89744edc271f/modules/scalacheck/src/weaver/scalacheck/CheckConfig.scala#L8

Reply:
Nice, thanks! I'll rewrite some tests tumorrow using that new config and see if that checks out (hope it doesnt take much more time without the parallelism)